.github/workflows/bcr_pr_review_approver.yml

```diff
@@ -0,0 +1,19 @@
+name: Review BCR Pull Requests
+on:
+  schedule:
+    - cron: "0 * * * *"    # Run this action hourly
+  workflow_dispatch:       # So that this can be triggered manually
+
+jobs:
+  review_prs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
+        with:
+          egress-policy: audit
+
+      - name: Run BCR PR Review Approver
+        uses: bazelbuild/continuous-integration/actions/bcr-pr-review-approver@b9b7f0dad8ab00a48cc262d1d38339a1dd11c7c5 # master
```

Comment:
I had a cursory look at https://github.com/bazelbuild/continuous-integration/blob/b9b7f0dad8ab00a48cc262d1d38339a1dd11c7c5/actions/bcr-pr-review-approver/index.js.

In L88, we are getting the author date from the latest commit and then proceed to consider the PR approved if any approval happened after this time. But the commit author date can be freely modified by the PR owner. I haven't tried this, but couldn't the owner just get an innocuous PR approved and then amend malicious changes into it, keeping the date constant?

Reply:
Hmm, nice catch! That's indeed a concern. Let me check if we can get the push time of a commit.

Reply:
My worry is that we'll miss other things. Is there someone on the OSS team at Google who could give this a proper security review?

Reply:
@mihaimaruseac Any advice? 😃

Reply:
We could do a go/gosst-review for this to look deeper if needed.

Just to check my understanding: this workflow is supposed to ensure that if a PR updates packages A and B, then it has been reviewed by an owner of A and an owner of B?

And the risk here is that owners of A approves for A and before an owner for B gets to review the PR author modifies package A?

For PRs with a single owner to review, having `Require approval of the most recent reviewable push` in branch protection should ensure that any change is reviewed by someone, but if we want 2 separate owners, maybe we need something to discard package A's review if a new commit lands to package A?

Reply:
Oh, I haven't thought about the patches. We could probably ask a BCR repo maintainer to review those PRs too (and rely only on package maintainers for other package reviews).

I think only nixpkgs matches the scenarios BCR has. But I'll need to see what processes for review they had.

Other package managers put all the responsibility on the package owners and have a team to react to mishaps after they occur (by yanking malicious packages), afaik.

Reply:
Of note: nearly every module has patches because of how the submit-to-bcr app works, mainly to change the version. https://github.com/bazelbuild/bazel-central-registry/issues/1566 might help get away from patches for that.

Reply:
I'll have to think more about this, I'll be back next week

Reply:
> I think only nixpkgs matches the scenarios BCR has. But I'll need to see what processes for review they had.

@aherrmann Do you know what's the review process for nixpkgs?

Reply:
The nixpkgs contribution and review process is documented [here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), see also [maintainer's README](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md).